add details that some customers may see modified project creation experience (#8108)
circleci/circleci-docs: jekyll/_cci2/getting-started.adoc

:page-description: A quickstart guide for adding a project to CircleCI and exploring some features
:page-liquid:

This quickstart guide takes you through connecting an existing project to CircleCI, using the in-app config editor, and inviting team members to collaborate on your new project.  

> NOTE: Starting June 14, 2023, some users may see a simplified experience when setting up a project on CircleCI.  See our link:https://discuss.circleci.com/t/product-update-simplifying-circleci-project-creation/48336[community forum] for more details.

Reviewer: ```suggestion
NOTE: Starting June 14, 2023, you may see a simplified experience when setting up a project on CircleCI.  See our link:https://discuss.circleci.com/t/product-update-simplifying-circleci-project-creation/48336[community forum] for more details.
```
